torchci/lib/bot/cancelWorkflowsOnCloseBot.ts

```diff
@@ -0,0 +1,45 @@
+import { Probot } from "probot";
+
+function cancelWorkflowsOnCloseBot(app: Probot): void {
+  app.on("pull_request.closed", async (ctx) => {
+    const owner = ctx.payload.repository.owner.login;
+    const repo = ctx.payload.repository.name;
+    const senderLogin = ctx.payload.sender.login;
+    const senderId = ctx.payload.sender.id;
+    const headSha = ctx.payload.pull_request.head.sha;
+
+    if (
+      senderLogin == "pytorchmergebot" ||
+      senderLogin == "pytorchbot" ||
+      senderLogin == "pytorch-bot[bot]" ||
+      senderId == 97764156 || // pytorchmergebot's id
+      senderId == 54816060 || // pytorch-bot's id
+      senderId == 21957446 // pytorchbot's id
+    ) {
+      return;
+    }
+
+    const workflowRuns = await ctx.octokit.actions.listWorkflowRunsForRepo({
+      owner,
+      repo,
+      head_sha: headSha,
+      per_page: 30,
+    });
+
+    await Promise.all(
+      workflowRuns.data.workflow_runs
```

Comment:
Consider checking the status of the worklowRuns to make sure the listWorkflow was successful

Reply:
afaict the way to catch errors for octokit is to put in a try catch, which im willing to do, but also we don't seem to do it anywhere else

Reply:
IMO, this is probably ok without a try/catch.  Even if the request fails, we will end up with what we currently have in which the workflow will continue running.